src/structure_managers/structure_manager.hh

```diff
@@ -911,7 +911,7 @@ namespace rascal {
 
     //! return the index of the atom/pair/etc. it is always the last one,
     //! since the other ones are accessed an Order above.
-    inline int get_atom_tag() const { return this->back(); }
+    inline decltype(auto) get_atom_tag() const { return this->back(); }
```

Comment:
Why not keep it an `int`, the lammps manager will point to it.

Reply:
auto is used here because the type returned is dictated by the storage rather than the function